src/internal/pachd/testpachd.go

```diff
@@ -65,6 +69,51 @@ func ActivateAuthOption(rootToken string) TestPachdOption {
 	}
 }
 
+// GetK8sClient is an option that sets k to the internal fake k8s client.  Mutations made to this
+// client will be visible to Pachyderm.
+func GetK8sClient(k *kubernetes.Interface) TestPachdOption {
+	return TestPachdOption{
+		MutatePachd: func(full *Full) {
+			*k = full.kubeClient
+		},
+	}
+}
+
+// WithS3Server runs the S3 gateway and sets the provided string pointer to its address.
+func WithS3Server(t *testing.T, addr *string) TestPachdOption {
+	l := testutil.Listen(t)
+	var server *http.Server
+	return TestPachdOption{
+		OnReady: func(ctx context.Context, full *Full) error {
+			full.s3Srv.Addr = l.Addr().String()
+			*addr = l.Addr().String()
+			server = full.s3Srv.Server
+			go full.s3Srv.Serve(l) //nolint:errcheck
```

Comment:
If you really, really wanted to you could do `go func() { doneCh <- full.s3Srv.Serve(l) }` and then read it down in the cleanup …